### ISSUE TITLE
Treat token references as connections (and show them in the Console)

### ISSUE THREAD

**Issue by eladb** [winglang/wing#6433]
### Use Case

When a resource references an attribute of another resource, it introduces a relationship between them, similarly to the relationship created when inflight code calls a method on another preflight resource.

I want to see these relationships (connections) in the Wing Console!

The fact that the relationship is introduced through a reference to an attribute and not a method call doesn't make it less of a relationship.

Before:
![image](https://github.com/winglang/wing/assets/598796/928bc49b-5e63-4574-b671-f1d5bb4c5967)

After:

![image](https://github.com/winglang/wing/assets/598796/53f29b4f-b4ef-43f7-b908-878e9b620516)

### Proposed Solution

We should identify tokens when we call `toSimulator()` in the resource configuration (simulator initially, but can also do it for terraform), and simply add a connection to the `tree.json`.

It should be straightforward to add something like this.

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.
* If this issue is labeled **needs-discussion**, it means the spec has not been finalized yet. Please reach out on the #dev channel in the [Wing Slack](https://t.winglang.io/slack).

**Comment by Chriscbr** [comment]
I wonder how this would be solved for Terraform, given that CDKTF tokens like `${TfToken.123}` often resolve into Terraform configuration tokens like `${aws_route_table.MyPrivateRouteTable.id}` (in HCL or JSON). It's possible the SDK could try and parse this string and reverse-engineer what logical construct it was associated with, but I wonder if there's a better way.

**Comment by eladb** [comment]
Yes, I would just do some parsing to identify the terraform level relationship.

For now, sim is good enough
